sklearn/linear_model/_glm/glm.py

```diff
@@ -263,13 +264,29 @@ def fit(self, X, y, sample_weight=None):
                 coef = self.coef_
             coef = coef.astype(loss_dtype, copy=False)
         else:
+            # Note: zero init for both the coefficients and the intercept
+            # is needed if we want the unpenalized model to converge to the
+            # minimum nornm solution which is likely to generalize better
+            # than other solutions.
+            shape = (n_features + 1,) if self.fit_intercept else (n_features,)
+            coef = np.zeros(shape, dtype=loss_dtype)
+
             if self.fit_intercept:
-                coef = np.zeros(n_features + 1, dtype=loss_dtype)
-                coef[-1] = linear_loss.base_loss.link.link(
-                    np.average(y, weights=sample_weight)
-                )
-            else:
-                coef = np.zeros(n_features, dtype=loss_dtype)
+                with warnings.catch_warnings():
+                    # Ignore log of zero warning
+                    warnings.filterwarnings("ignore", category=RuntimeWarning)
+                    valid_zero_intercept = self._base_loss.in_y_pred_range(
+                        self._base_loss.link.link([0.0])
+                    )
+                    if not valid_zero_intercept:
+                        # Initialize the intercept to a small positive value to
+                        # ensure that so
+                        # linear_loss.base_loss.link.link(raw_prediction) is in
+                        # the admissible range of the distribution family /
+                        # link function (in particular for Tweedie with p>=1
+                        # with identity link) to be able to compute the first
+                        # gradient.
+                        coef[-1] = 64 * np.finfo(loss_dtype).eps
```

Comment:
The magic 64 again :wink:

Reply:
I think it works for a variety of small eps but getting too close to the minimum eps would make convergence less numerically stable for some combinations of family / link.